### ISSUE TITLE
rosdep installs gcc on OS X

### ISSUE THREAD

**Issue by NikolausDemmel** [ros-infrastructure/rosdep#328]
With recent changes in homebrew, the `gfortran` formula was merged with the `gcc` formula. This means, that the rosdep key `eigen` is now resolved (via gfortran) to install `gcc` on osx/homebrew. This is a problem, since after installing gcc, in particular the `c++` executable on the PATH is `/usr/local/bin/c++`, i.e. gcc, rather than clang. This means that (some) ROS packages get compiled with gcc, which in turn causes linker errors with third party libraries due to the libc++/libstdc++ incompatibilities.

I don't know how to resolve this. Does eigen desperately need gfortran? At least the homebrew formula does not have that dependency. If gfortran is really needed (the only other reference is python-scypi on osx/brew), we need to figure out a solution that does not pull in gcc, maybe provide the old standalone gfortran formula ourselves?

TL;DR: eigen should not cause gcc to be installed on OS X.
